connectors/sources/mongo.py

```diff
@@ -87,10 +87,34 @@ def _serialize(value):
         return doc
 
     async def get_docs(self, filtering=None):
+        await self._validate_configuration()
+
+        db = self.client[self.configuration["database"]]
+
         logger.debug("Grabbing collection info")
-        collection = self.db[self.configuration["collection"]]
+        collection = db[self.configuration["collection"]]
 
         async for doc in collection.find():
             yield self.serialize(doc), None
 
         self._dirty = False
+
+    async def _validate_configuration(self):
```

Comment:
Do you think it would be useful to have a `validate_configuration` method inside `BaseDataSource` and implement it everywhere in our native connectors? We could also call it somewhere in our framework instead of relying on the connector implementations. As other configuration concerns like `get_default_configuration` and instantiating the `DataSourceConfiguration` lie inside the `BaseDataSource` it would feel a bit more natural IMO. Just thinking out loud

Reply:
I think that was the original Tarek's proposal, wanted to double check it with him :)